lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -121,39 +122,77 @@ public function convertException($message, DriverException $exception)
         return new Exception\DriverException($message, $exception);
     }
 
+
     /**
      * {@inheritdoc}
      */
     public function createDatabasePlatformForVersion($version)
     {
-        if ( ! preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
-            throw DBALException::invalidPlatformVersionSpecified(
-                $version,
-                '<major_version>.<minor_version>.<patch_version>'
-            );
-        }
+
+        // Detect vendor and version
 
         if (false !== stripos($version, 'mariadb')) {
-            return $this->getDatabasePlatform();
-        }
+            // MariaDB does not follow one rule for versions ;)
 
-        $majorVersion = $versionParts['major'];
-        $minorVersion = isset($versionParts['minor']) ? $versionParts['minor'] : 0;
-        $patchVersion = isset($versionParts['patch']) ? $versionParts['patch'] : null;
 
-        if ('5' === $majorVersion && '7' === $minorVersion && null === $patchVersion) {
-            $patchVersion = '9';
-        }
 
-        $version = $majorVersion . '.' . $minorVersion . '.' . $patchVersion;
+            // Will catch things like '10.2.8-MariaDB-1~lenny-log'
+            preg_match('/(?P<major>\d{1,2})\.(?P<minor>\d{1,2})\.(?P<patch>\d+)/', $version, $firstMatch);
+            // Will catch things like '5.5.5-10.2.8-MariaDB-10.2.8+maria~xenial-log'
+            preg_match('/\-(?P<major>10)\.(?P<minor>\d{1,2})\.(?P<patch>\d+)/', $version, $secondMatch);
```

Comment:
Which of the `'10.2.8'` is this supposed to match? The one after the `'5.5.5'` or after `'MariaDB'`? Why only the major version of `10` is considered?

Reply:
Both expressions are almost the same with the exception of the leading dash. Could they be combined together?

Reply:
Maybe, but for now I would consider to create a separate class or method (will be needed for `mysqli` support in MysqliConnection)... Yes I'm not happy with detection, but wanted something to start. It's in the todo. Once a better detection is ready I'll need your feedback.